rust/lance-linalg/src/distance/dot.rs

```diff
@@ -303,4 +310,58 @@ mod tests {
         let y: Vec<f64> = (120..140).map(|v| f64::from_i32(v).unwrap()).collect();
         assert_eq!(Float64Type::dot(&x, &y), dot(&x, &y));
     }
+
+    /// Reference implementation of dot product.
+    fn dot_scalar_ref(x: &[f64], y: &[f64]) -> f32 {
+        x.iter().zip(y.iter()).map(|(&x, &y)| x * y).sum::<f64>() as f32
+    }
+
+    // Accuracy of dot product depends on the size of the components
+    // of the vector.
+    fn max_error(x: &[f64], y: &[f64]) -> f32 {
+        let dot = x
+            .iter()
+            .cloned()
+            .zip(y.iter().cloned())
+            .map(|(x, y)| x.abs() * y.abs())
+            .sum::<f64>();
+        (1e-6_f64 * dot) as f32
+    }
```

Comment:
Does it depend on the size of the components of the dimension of the vector?  I guess it's not intuitive why `sum(x*y)` is correct here.

Reply:
I pushed a new version with better comments explaining what is going on.